src/librustc/sir.rs

```diff
@@ -102,7 +104,74 @@ impl SirFuncCx {
 
     /// Converts a MIR statement to SIR, appending the result to `bb`.
     pub fn codegen_statement(&mut self, bb: ykpack::BasicBlockIndex, stmt: &mir::Statement<'_>) {
-        // FIXME: Nothing is implemented yet.
-        self.push_stmt(bb, ykpack::Statement::Unimplemented(format!("{:?}", stmt)));
+        match stmt.kind {
+            mir::StatementKind::Assign(box (ref place, ref rvalue)) => {
+                let assign = self.lower_assign_stmt(place, rvalue);
+                self.push_stmt(bb, assign);
+            },
+            _ => self.push_stmt(bb, ykpack::Statement::Unimplemented(format!("{:?}", stmt)))
+        }
+    }
+
+    fn lower_assign_stmt(&self, lvalue: &mir::Place<'_>, rvalue: &mir::Rvalue<'_>) -> ykpack::Statement {
+        let lhs = ykpack::Local(lvalue.local.as_u32());
```

Comment:
Shouldn't this be `lower_local()` now?

Reply:
Oh I missed that one. Also, shouldn't we actually call it `lower_place`?

Reply:
A place and a local are different concepts. Possible confusion?

Reply:
Okay, then maybe I should pass in the local and not the place, e.g. `lower_local(place.local)`.

Reply:
Yep. I'd expect `lower_local()` to accept a MIR local and return a SIR local. It shouldn't be to do with place at all.

Reply:
Done.